api/src/controllers/DashboardController.ts

```diff
@@ -0,0 +1,28 @@
+import { NextFunction, Request, Response } from "express";
+import ErrorController from "./ErrorController";
+import DashboardService from "../services/DashboardService";
+
+export default class DashboardController extends ErrorController {
+  private readonly dashboardService = new DashboardService();
+
+  public async getDashboardStats(
+    _: Request,
+    res: Response,
+    next: NextFunction
+  ) {
+    try {
+      const dashboardStats = {
+        totalInventoryCount:
+          await this.dashboardService.getTotalInventoryCount(),
+        totalInventoryStatuses:
+          await this.dashboardService.getTotalInventoryStatus(),
+        recentlyAddedInventory:
+          await this.dashboardService.getRecentlyAddedInventory(),
```

Comment:
C: Use `Promise.all`, or ensure that these promises are able to run in parallel.

Reply:
Might be worth adding a call to pre-create the database connections, so it doesn't try and create 3 in parallel.